test/interval.jl

```diff
@@ -144,6 +143,164 @@
         @test span(interval) == Hour(3)
     end
 
+    @testset "maximum/minimum" begin
+        # Helper functions that manage the value we should be expecting from min and max.
+        function _min_val_helper(interval, a, unit)
+            t = eltype(interval)
+            # If the interal is empty, min is nothing
+            isempty(interval) && return nothing
+
+            # If a is in the interval, it is closed/unbounded and min is the first value.
+            # If a is nothing then it is unbounded and min is typemin(T)
+            a === nothing && return typemin(t)
+            a ∈ interval && return first(interval)
+
+            # From this point on, b ∉ interval so the bound is Open
+            # Also, if a is infinite we return typemin
+            # If it's an abstractfloat, we can't return just typemin since typemin IS Inf and
+            # since the bound is open at this point, Inf ∉ interval So we return the one after INF
+            !isfinite(a) && t <: AbstractFloat && return nextfloat(a)
+            !isfinite(a) && return typemin(t)
+
+            f = first(interval)
+            nv = if t <: AbstractFloat && (unit === nothing || unit == eps(t))
+                nextfloat(f)
+            else
+                f + unit
+            end
+
+            nv ∈ interval && return nv
+
+            # If we get to this point, the min/max functions throw a DomainError
+            # Since we want our tests to be predictable, we will not throw an error in this helper.
+        end
+
+        function _max_val_helper(interval, b, unit)
+            t = eltype(interval)
+            # If the interal is empty, min is nothing
+            isempty(interval) && return nothing
+
+            # If a is in the interval, it is closed/unbounded and min is the first value.
+            # If a is nothing then it is unbounded and min is typemin(T)
+            b === nothing && return typemax(t)
+            b ∈ interval && return last(interval)
+
+            # From this point on, b ∉ interval so the bound is Open
+            # Also, if a is infinite we return typemin
+            # If it's an abstractfloat, we can't return just typemin since typemin IS Inf and
+            # since the bound is open at this point, Inf ∉ interval So we return the one after INF
+            !isfinite(b) && t <: AbstractFloat && return prevfloat(b)
+            !isfinite(b) && return typemax(t)
+
+            l = last(interval)
+            nv = if t <: AbstractFloat && (unit === nothing || unit == eps(t))
+                prevfloat(l)
+            else
+                l - unit
+            end
+
+            nv ∈ interval && return nv
+
+            # If we get to this point, the min/max functions throw a DomainError
+            # Since we want our tests to be predictable, we will not throw an error in this helper.
+        end
+        @testset "bounded intervals" begin
+            bounded_test_vals = [
```

Comment:
Can't use Infinity.jl since `isfinite`does ` ∞ -  ∞` which causes an error. Could make a workaround for this

Reply:
If you rebase your changes this should no longer be an issue:
```julia
julia> isfinite(∞)
false

julia> Intervals.isfinite(∞)
false
```

Reply:
Might need some more clarification on this.

I rebased in the traditional sense `git rebase master` and I also updated from master, but it still doesn't work for some reason. 🙁 

If I replicate your calls in the REPL, it works fine as well.

Reply:
```julia
julia> using Dates,Infinity, Intervals

julia> x = Interval{Open, Open}(Date(2013, 2, 13), ∞)
Interval{InfExtendedTime{Date},Open,Open}(InfExtendedTime{Date}(Date("2013-02-13")), InfExtendedTime{Date}(∞))

julia> minimum(x, increment=Day(1))
InfExtendedTime{Date}(Date("2013-02-14"))

julia> maximum(x, increment=Day(1))
ERROR: ∞-∞ is undefined
Stacktrace:
 [1] + at /Users/arvindmaan/.julia/packages/Infinity/ezs8D/src/infinite/arithmetic.jl:9 [inlined]
 [2] -(::Infinite, ::Infinite) at /Users/arvindmaan/.julia/packages/Infinity/ezs8D/src/infinite/arithmetic.jl:10
 [3] -(::InfExtendedTime{Date}, ::InfExtendedTime{Date}) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Dates/src/arithmetic.jl:9
 [4] isfinite(::InfExtendedTime{Date}) at /Users/arvindmaan/.julia/dev/Intervals/src/isfinite.jl:3
 [5] maximum(::Interval{InfExtendedTime{Date},Open,Open}; increment::Day) at /Users/arvindmaan/.julia/dev/Intervals/src/interval.jl:228
 [6] top-level scope at REPL[7]:1
```

Reply:
The trouble is definitely from `InfExtendedTime` specifically here. I'll see if I can come up with a solution for that

Reply:
Add the following to the end of `src/isfinite.jl` and everything should be working:
```julia
isfinite(x::Union{Type{T}, T}) where T<:TimeType = Base.isfinite(x)
```